x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/CentroidCalculator.java

```diff
@@ -246,22 +246,20 @@ private void visitPoint(double x, double y) {
         }
 
         private void visitLine(int length, CoordinateSupplier x, CoordinateSupplier y) {
-            // check line has length
-            double originDiffX = x.get(0) - x.get(1);
-            double originDiffY = y.get(0) - y.get(1);
-            if (originDiffX != 0 || originDiffY != 0) {
-                // a line's centroid is calculated by summing the center of each
-                // line segment weighted by the line segment's length in degrees
-                for (int i = 0; i < length - 1; i++) {
-                    double diffX = x.get(i) - x.get(i + 1);
-                    double diffY = y.get(i) - y.get(i + 1);
-                    double xAvg = (x.get(i) + x.get(i + 1)) / 2;
-                    double yAvg = (y.get(i) + y.get(i + 1)) / 2;
-                    double weight = Math.sqrt(diffX * diffX + diffY * diffY);
+            // a line's centroid is calculated by summing the center of each
+            // line segment weighted by the line segment's length in degrees
+            for (int i = 0; i < length - 1; i++) {
+                double diffX = x.get(i) - x.get(i + 1);
+                double diffY = y.get(i) - y.get(i + 1);
+                double xAvg = (x.get(i) + x.get(i + 1)) / 2;
+                double yAvg = (y.get(i) + y.get(i + 1)) / 2;
+                double weight = Math.sqrt(diffX * diffX + diffY * diffY);
+                if (weight == 0) {
+                    // degenerated line, it can be considered a point
+                    visitPoint(x.get(i), y.get(i));
```

Comment:
Do we believe this is right?

If a line segment has repeating sections where the weight is effectively zero, then we could potentially be counting multiple points per `Line`?

Reply:
I think it is ok. If there is one segment with length, the centroid of the points will be discarded. If all points are equivalent, then it will be weight by the number of points. I think it is the right thing to do.

Reply:
OK. just wanted to be sure! thanks